src/main/java/org/rumbledb/context/Name.java

```diff
@@ -0,0 +1,214 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Authors: Ghislain Fourny
+ *
+ */
+
+package org.rumbledb.context;
+
+import java.io.Serializable;
+
+import org.rumbledb.exceptions.OurBadException;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.KryoSerializable;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+
+/**
+ * This class represents expanded names, corresponding to QNames in the W3C XQuery standard.
+ * 
+ * An expanded name consists logically of a namespace (possibly absent) and a local name.
+ * 
+ * A namespace may also have a proxy called a namespace prefix, used in lexical representation
+ * (e.g., name of the function or variable in JSONiq code).
+ * 
+ * Expanded names serve as (global or local) variable names, function names, and error codes.
+ * 
+ * @author Ghislain Fourny
+ *
+ */
+public class Name implements Comparable<Name>, Serializable, KryoSerializable {
+    private static final long serialVersionUID = 1L;
+    private String namespace;
+    private String prefix;
+    private String localName;
+    private static final String RUMBLE_NS = "http://rumbledb.org/main-namespace";
+    public static final Name CONTEXT_ITEM = createVariableInNoNamespace("$");
+    public static final Name CONTEXT_POSITION = createVariableInNoNamespace("$position");
+    public static final Name CONTEXT_COUNT = createVariableInNoNamespace("$count");
+
+    private Name() {
+        this.namespace = null;
+        this.prefix = null;
+        this.localName = null;
+    }
+
+    private Name(String namespace, String prefix, String localName) {
+        this.namespace = namespace;
+        this.prefix = prefix;
+        this.localName = localName;
+        if (this.prefix != null && this.namespace == null) {
+            throw new OurBadException("Namespace is null, but prefix is present");
+        }
+    }
+
+    /**
+     * Creates an expanded name that has no namespace. In JSONiq, unprefixed variables are never in
+     * a namespace.
+     * 
+     * @param localName the name of the variable
+     * @return the expanded name
+     */
+    public static Name createVariableInNoNamespace(String localName) {
+        return new Name(null, null, localName);
+    }
+
+    /**
+     * Creates an expanded name that has the Rumble namespace. By default, in Rumble, unprefixed
+     * function names live in the Rumble namespace. This namespace is for convenience and includes
+     * all functions in XQuery's fn namespace, JSONiq's jn and jnlib namespaces, as well as any
```

Comment:
If jn and jnlib are abbreviations, I don't think I am familiar with them. It could be helpful to give the full name in that case.

Reply:
Added